cylc/uiserver/schema.py

```diff
@@ -287,9 +289,50 @@ async def get_elements(root, info, **kwargs):
     kwargs['exworkflows'] = [
         Tokens(w_id) for w_id in kwargs['exworkflows']]
 
+    # ret = []
+    # if resource fields requested
+    # ret.append(get_job_resource_request(info, kwargs))
+    # if database fields requested
+    # ret.append(await list_elements(kwargs))
+    # merge results if both sets of fields were requested
+    # return merge_results(ret)
     return await list_elements(kwargs)
 
 
+def get_job_resource_request(info, args):
+    """Return the resource request for a task as configured.
+    TODO:
+    * Determine the batch system by mapping the task onto the specified
+      platform.
+    * We will likely want the resource requested by a task when it was
+      submitted (which may be altered by reloads or broadcasts) rather than the
+      resource that is configured in the workflow.
+    """
+
+    ret = {}
+
+    for workflow_id in args['workflows']:
+
+        for namespace in args['tasks']:
+            namespace_id = (Tokens(workflow_id).duplicate(
+                namespace=namespace).id)
+
+            namespace = info.context['resolvers'].data_store_mgr.data[
+                workflow_id][TASKS][namespace_id]
+
+            directives = {item['key']: item['value']
+                          for item in json.loads(namespace.runtime.directives)}
+
+            memory = int(directives.get('--mem', 0))
+            cpus = math.ceil(int(directives.get('--tasks', 0)) / 2)
+            time = float(directives.get('--time', 0))
+
+            # TODO: put the results into the format GraphQL is expecting
+            ret[...] = {'memory': memory, 'cpus': cpus, 'time': time}
+    breakpoint()
+    return ret
```

Comment:
Strongly recommend keeping these PRs to the profiler functionality alone so we can get that functionality released and in use more quickly.

We can come back to requested resource in a later PR.

Reply:
Done